sui_core/src/authority/authority_store.rs

```diff
@@ -303,14 +307,50 @@ impl<
             .collect())
     }
 
+    // Returns true if the given version of the object has not been deleted (or wrapped).
+    fn object_is_alive(&self, object_id: &ObjectID) -> Result<bool, SuiError> {
+        let obj = self
+            .parent_sync
+            .iter()
+            .skip_prior_to(&(*object_id, SequenceNumber::MAX, ObjectDigest::MAX))?
+            .next();
+
+        Ok(obj.map(|(k, _)| k.0 == *object_id && k.2.is_alive()).unwrap_or(false))
+    }
+
     /// Read an object and return it, or Err(ObjectNotFound) if the object was not found.
     pub fn get_object(&self, object_id: &ObjectID) -> Result<Option<Object>, SuiError> {
-        self.objects.get(object_id).map_err(|e| e.into())
+        let obj = self
+            .objects
+            .iter()
+            .skip_prior_to(&ObjectKey::max_for_id(object_id))?
+            .next();
+
+        // Note that the two reads in this function are (obviously) not atomic, and the
+        // object may be deleted after we have read it. Hence we check object_is_alive last
+        // so that the write to self.parent_sync gets the last word.
+        match obj {
+            Some((ObjectKey(obj_id, _), obj)) if obj_id == *object_id => {
+                if self.object_is_alive(&obj_id)? {
+                    Ok(Some(obj))
+                } else {
+                    Ok(None)
+                }
+            }
+            _ => {
+                error!(?object_id, "Object is missing parent_sync entry, data store is inconsistent");
```

Comment:
The error log shouldn't be here, it should be normal when an object is just not in the store. I think what you are looking for is the None above with the Some case.

Reply:
sorry - i think the lack of sleep from the new baby is affecting me more than I realize